components/renku_data_services/session/api.spec.yaml

```diff
@@ -531,11 +641,96 @@ components:
       description: A description for the resource
       type: string
       maxLength: 500
+    ImageBuilder:
+      description: A container image builder
+      type: object
+      additionalProperties: false
+      properties:
+        builder_id:
+          $ref: "#/components/schemas/Ulid"
+        repository:
+          $ref: "#/components/schemas/Repository"
+        revision:
+          $ref: "#/components/schemas/Revision"
+        subdir:
+          $ref: "#/components/schemas/ImageBuilderSubDir"
+        venv_kind:
+          $ref: "#/components/schemas/VEnvKind"
+        frontend_kind:
+          $ref: "#/components/schemas/FrontendKind"
+      required:
+        - repository
+        - venv_kind
+        - frontend_kind
+      example:
+        builder_id: 01AN4Z79ZS6XX96588FDX0H099
+        container_image: my_project_name/environment_name:mytag
+        repository: https://gitlab.com/my_namespace/my_project_repo.git
+        revision: main
+        subdir: "."
+        venv_kind: conda
+        frontend_kind: jupyterlab
+    ImageBuilderPost:
+      description: Data required to create an image builder
+      type: object
+      properties:
+        builder_id:
+          $ref: "#/components/schemas/Ulid"
+        repository:
+          $ref: "#/components/schemas/Repository"
+        revision:
+          allOf:
+            - $ref: "#/components/schemas/Revision"
+            - default: main
+        subdir:
+          allOf:
+            - $ref: "#/components/schemas/ImageBuilderSubDir"
+            - default: /
+        venv_kind:
+          $ref: "#/components/schemas/VEnvKind"
+        frontend_kind:
+          $ref: "#/components/schemas/FrontendKind"
+      required:
+        - repository
+        - venv_kind
+        - frontend_kind
+    ImageBuilderPatch:
+      description: Update an image builder
+      type: object
+      additionalProperties: false
+      properties:
+        builder_id:
+          $ref: "#/components/schemas/Ulid"
+        repository:
+          $ref: "#/components/schemas/Repository"
+        revision:
+          $ref: "#/components/schemas/Revision"
+        subdir:
+          $ref: "#/components/schemas/ImageBuilderSubDir"
+        venv_kind:
+          $ref: "#/components/schemas/VEnvKind"
+        frontend_kind:
+          $ref: "#/components/schemas/FrontendKind"
     ContainerImage:
       description: A container image
       type: string
       maxLength: 500
       example: renku/renkulab-py:3.10-0.18.1
+    Repository:
+      description: The repository used as a basis for a custom builder
+      type: string
+      maxLength: 500
+      example: "https://github.com/voila-gallery/gaussian-density"
+    Revision:
+      description: The revision of the repository used as a basis for a custom builder
```

Comment:
```suggestion
      description: The revision of the repository used as a basis for a custom builder. This can be a git branch, a git tag or a git commit.
```